apps/api/src/domain/models/photograph/repositories/arango-photograph-query-repository.ts

```diff
@@ -0,0 +1,188 @@
+    ) {
+        this.database = new ArangoDatabaseForCollection(
+            new ArangoDatabase(arangoConnectionProvider.getConnection()),
+            'photograph__VIEWS'
+        );
+    }
+
+    async fetchById(id: AggregateId): Promise<Maybe<PhotographViewModel>> {
+        const documentSearchResult = await this.database.fetchById(id);
+
+        if (isNotFound(documentSearchResult)) {
+            return documentSearchResult;
+        }
+
+        const viewModelDto = mapDatabaseDocumentToEntityDto(
+            documentSearchResult
+        ) as PhotographViewModel & {
+            actions: ICommandFormAndLabels[];
+        };
+
+        return PhotographViewModel.fromDto(viewModelDto);
+    }
+
+    async fetchMany(): Promise<PhotographViewModel[]> {
+        const documents = await this.database.fetchMany();
+
+        const viewModelsFromRepo = documents.map((doc) =>
+            PhotographViewModel.fromDto(mapDatabaseDocumentToEntityDto(doc))
+        ) as PhotographViewModel[];
+
+        return viewModelsFromRepo;
+    }
+
+    async count(): Promise<number> {
+        return this.database.getCount();
+    }
+
+    async create(view: PhotographViewModel): Promise<void> {
+        const viewToCreate = mapEntityDtoToDatabaseDocument(view);
+
+        // TODO If we're going to throw here, we need to wrap the top level event handlers in a try...catch
+        return this.database.create(viewToCreate).catch((error) => {
+            throw new InternalError(
+                `failed to create photograph view in ArangoPhotographQueryRepository`,
+                [error]
+            );
+        });
+    }
+
+    async createMany(views: PhotographViewModel[]): Promise<void> {
+        return this.database
+            .createMany(views.map(mapEntityDtoToDatabaseDocument))
+            .catch((error) => {
+                throw new InternalError(
+                    `failed to create many photograph views in ArangoPhotographQueryRepository`,
+                    [error]
+                );
+            });
+    }
+
+    async publish(id: AggregateId): Promise<void> {
+        const query = `
+        FOR doc IN @@collectionName
+        FILTER doc._key == @id
+        UPDATE doc WITH {
+            isPublished: true,
+            actions: REMOVE_VALUE(doc.actions, "PUBLISH_RESOURCE")
+        } IN @@collectionName
+         `;
+
+        const bindVars = {
+            '@collectionName': 'photograph__VIEWS',
+            id: id,
+        };
+
+        const cursor = await this.database
+            .query({
+                query,
+                bindVars,
+            })
+            .catch((reason) => {
+                throw new InternalError(
+                    `Failed to publish photograph via PhotographRepository: ${reason}`
+                );
+            });
+
+        await cursor.all();
+    }
+
+    async allowUser(photographId: AggregateId, userId: AggregateId): Promise<void> {
```

Comment:
Let's look at how to share code between the query repos once you're all done with the current PR. I think we can cut the work down to half or even less if we do this.